docs/aws/audit/snsmonitoring/rules/is_topic_subscription_public.mdx

```diff
@@ -23,6 +23,251 @@ HITRUST, AWSWAF, SOC2, NISTCSF, PCIDSS
+
+4. Check the `Policy` attribute of each SNS topic. If the `Policy` attribute contains a statement with `"Effect": "Allow"` and `"Principal": "*"`, it means that the SNS topic allows global subscribe. This is a misconfiguration and should be fixed. You can use a Python script to automate this process. Here is a simple example:
+   ```python
+   import json
+   import subprocess
+
+   # Get the list of all SNS topics
+   topics = json.loads(subprocess.check_output(['aws', 'sns', 'list-topics']))['Topics']
+
+   for topic in topics:
+       # Get the attributes of each SNS topic
+       attributes = json.loads(subprocess.check_output(['aws', 'sns', 'get-topic-attributes', '--topic-arn', topic['TopicArn']]))['Attributes']
+       
+       # Check the policy of each SNS topic
+       policy = json.loads(attributes['Policy'])
+       for statement in policy['Statement']:
+           if statement['Effect'] == 'Allow' and statement['Principal'] == '*':
+               print(f"The SNS topic {topic['TopicArn']} allows global subscribe.")
+   ```
+   This script will print the ARNs of all SNS topics that allow global subscribe.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing Python scripts to interact with AWS, you need to install the necessary libraries. The primary library you need is boto3, which is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Set up AWS credentials: Boto3 needs your AWS credentials (access key and secret access key) to interact with AWS services. You can set them up in your environment variables:
+
+   ```python
+   import os
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_access_key'
+   ```
+
+3. Write a Python script to list all SNS topics and their policies: You can use the `list_topics` method to get all SNS topics and then use the `get_topic_attributes` method to get the policy of each topic. Here is a sample script:
+
+   ```python
+   import boto3
+
+   sns = boto3.client('sns')
+
+   # List all SNS topics
+   response = sns.list_topics()
+   topics = response['Topics']
+
+   for topic in topics:
+       # Get the policy of each topic
+       topic_arn = topic['TopicArn']
+       response = sns.get_topic_attributes(TopicArn=topic_arn)
+       policy = response['Attributes']['Policy']
+       print(f'Topic ARN: {topic_arn}\nPolicy: {policy}\n')
+   ```
+
+4. Analyze the policies: The policy of each SNS topic is a JSON string. You can convert it to a Python dictionary using the `json.loads` method and then check if it allows global subscribe. If the "Principal" in the policy is "*", it means the topic allows global subscribe. Here is how you can modify the above script to detect such topics:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>